### PR TITLE
New way add cellStyle && rowStyle && attr && fixes

### PR DESCRIPTION
**Pull request by andig89**
TL;DR
1. Has own styles and attr:
    - header
    - tr in tbody
    - td in tbody
    - special colspan and rowspan options which automatic spans col/row
2. Defaults options
3. add replaceOptionsDefaults option
4. Toggle options DOESN'T WORK!! I didn't fix it.
5. fix updateByUniqueId - add replace option
6. fix updateRow - add replace option

Long version:
//sort not working only in jsfiddle. I wasn't looking for a solution to fix this.
columnsOptionsDefaults<column<row
for attr rowspan/colspan:
rowsOptionsDefaultscolumnsOptionsDefaults<column<row

New options: //case sensitive!
    - css
    - class
    - attrHtml

http://jsfiddle.net/andig89/kc1cp8un/
```
//...//    
    rowsOptionsDefaults: {
        // also we can add
        // class: '.test'
        // attrHtml: {
        //    onclick: 'alert("I work :)!")'
        // },
        // rowspan: 2,
        // colspan: 2,
        // and any other
        css: {
            color: 'orange'
        },
        rows: {
            1: { // index row (start from 0)
                css: {
                    'font-size': '20px'
                }
                // also we can add
                // class: '.test'
                // attrHtml: {
                //    onclick: 'alert("I work :)!")'
                // },
                // rowspan: 2,
                // colspan: 2,
                // and any other
            },
            2: { // index row (start from 0)
                head: {
                    // also we can add
                    // class: '.test'
                    // attrHtml: {
                    //    onclick: 'alert("I work :)!")'
                    // },
                    // rowspan: 2,
                    // colspan: 2,
                    // and any other
                    css: {
                        color: 'blue'
                    }
                }
            }
        }
    },
    columns: [
        [{
            // also we can add
            // class: '.test'
            // attrHtml: {
            //    onclick: 'alert("I work :)!")'
            // },
            // rowspan: 2,
            // colspan: 2,
            // and any other
            title: 'Table',
            head: {
                // also we can add
                // class: '.test'
                // attrHtml: {
                //    onclick: 'alert("I work :)!")'
                // },
                // rowspan: 2,
                // colspan: 2,
                // and any other
                colspan: 5
            },
            unsedDataColumn: true,
            align: 'center'
        }],
        [{
            field: 'state',
            checkbox: true,
            rows: {
                1: {
                    head: {
                        rowspan: 2
                    }
                }
            },
            align: 'center',
            valign: 'middle'
        }, {
            title: 'Item ID',
            field: 'id',
            rows: {
                0: {
                    css: {
                        color: 'red'
                    }
                }
            },
            head: {
                rowspan: 2
            },
            align: 'center',
            valign: 'middle',
            sortable: true,
            footerFormatter: totalTextFormatter
        }, {
            title: 'Item Detail',
            head: {
                colspan: 3
            },
            unsedDataColumn: true,
            align: 'center'
        }],
        [{
            field: 'name',
            title: 'Item Name',
            sortable: true,
            editable: true,
            footerFormatter: totalNameFormatter,
            align: 'center'
        }, {
            field: 'price',
            title: 'Item Price',
            sortable: true,
            align: 'center',
            editable: {
                type: 'text',
                title: 'Item Price',
                validate: function(value) {
                    value = $.trim(value);
                    if (!value) {
                        return 'This field is required';
                    }
                    if (!/^\$/.test(value)) {
                        return 'This field needs to start width $.'
                    }
                    var data = $table.bootstrapTable('getData'),
                        index = $(this).parents('tr').data('index');
                    console.log(data[index]);
                    return '';
                }
            },
            footerFormatter: totalPriceFormatter
        }, {
            field: 'operate',
            title: 'Item Operate',
            align: 'center',
            events: operateEvents,
            formatter: operateFormatter
        }]
    ]
//...//
```

http://jsfiddle.net/andig89/mewn1s0p/
```
//...//
    columnsOptionsDefaults: {
        // also we can add
        // class: '.test'
        // attrHtml: {
        //    onclick: 'alert("I work :)!")'
        // },
        // rowspan: 2,
        // colspan: 2,
        // and any other
        css: {
            color: 'green'
        },
        colspan: 1 // rowsOptionsDefaultscolumnsOptionsDefaults<column<row try to remove
    },
    rowsOptionsDefaults: {
        css: {
            color: 'orange'
        },
        rows: {
            0: {
                css: {
                    'font-size': '20px'
                }
            },
            1: {
                css: {
                    'font-size': '20px'
                }
            },
            2: {
                head: {
                    css: {
                        color: 'blue'
                    }
                }
            },
            3: {
                colspan: 2
            }
        }
    },
    columns: [
        [{
            // also we can add
            // class: '.test'
            // attrHtml: {
            //    onclick: 'alert("I work :)!")'
            // },
            // rowspan: 2,
            // colspan: 2,
            // and any other
            title: 'Table',
            head: {
                // also we can add
                // class: '.test'
                // attrHtml: {
                //    onclick: 'alert("I work :)!")'
                // },
                // rowspan: 2,
                // and any other
                colspan: 5
            },
            unsedDataColumn: true,
            align: 'center'
        }],
        [{
            field: 'state',
            checkbox: true,
            rows: {
                1: {
                    // also we can add
                    // class: '.test'
                    // attrHtml: {
                    //    onclick: 'alert("I work :)!")'
                    // },
                    // rowspan: 2,
                    // colspan: 2,
                    // and any other
                    head: {
                        // also we can add
                        // class: '.test'
                        // attrHtml: {
                        //    onclick: 'alert("I work :)!")'
                        // },
                        // colspan: 2,
                        // and any other
                        rowspan: 2,
                        attrHtml: {
                            onclick: 'alert("I work :)!")'
                        }
                    }
                }
            },
            align: 'center',
            valign: 'middle'
        }, {
            title: 'Item ID',
            field: 'id',
            css: {
                height: '60px'
            },
            rows: {
                0: {
                    css: {
                        color: 'red'
                    }
                },
                3: {
                    replaceOptionsDefaults: true, // use '=' instead of $.extend
                    css: {
                        height: '40px',
                        'font-weight': 'bold'
                    }
                }
            },
            head: {
                rowspan: 2
            },
            align: 'center',
            valign: 'middle',
            sortable: true,
            footerFormatter: totalTextFormatter
        }, {
            title: 'Item Detail',
            head: {
                colspan: 3
            },
            unsedDataColumn: true,
            align: 'center'
        }],
        [{
            field: 'name',
            title: 'Item Name',
            sortable: true,
            editable: true,
            footerFormatter: totalNameFormatter,
            align: 'center'
        }, {
            field: 'price',
            title: 'Item Price',
            sortable: true,
            align: 'center',
            editable: {
                type: 'text',
                title: 'Item Price',
                validate: function(value) {
                    value = $.trim(value);
                    if (!value) {
                        return 'This field is required';
                    }
                    if (!/^\$/.test(value)) {
                        return 'This field needs to start width $.'
                    }
                    var data = $table.bootstrapTable('getData'),
                        index = $(this).parents('tr').data('index');
                    console.log(data[index]);
                    return '';
                }
            },
            footerFormatter: totalPriceFormatter
        }, {
            field: 'operate',
            title: 'Item Operate',
            align: 'center',
            events: operateEvents,
            formatter: operateFormatter
        }]
    ]
//...//
```

### CODE REVIEW

**Comment by gpedro** [comment]
redefining columnDefaults not solve your problem? look #3237

**Comment by wenzhixin** [comment]
invalid PR
